### ISSUE TITLE
Duplicated dependencies

### ISSUE THREAD

**Issue by 0x4E0x650x6F** [RIPE-NCC/whois#330]
Tested on Slackware 14.1 and Debian 7, on oracle 1.8.71 and maven 3. 
After incial issue compiling the project flowing the guide at https://github.com/RIPE-NCC/whois/wiki/Installation-instructions. 
Their were serveral warnings in the compilation process referring to duplicated dependencies. 
![duplicated_libraries](https://cloud.githubusercontent.com/assets/1144303/12674578/dfedfa0e-c67b-11e5-9028-87b17b96a447.png)

**Comment by eshryane** [comment]
As project dependencies themselves include their own dependencies, sometimes duplicates occur.

This results in a warning from the maven-shade-plugin when running the release profile:
```
mvn clean install -Prelease
```

In this case, there is a top-level dependency on log4j, but slf4j also depends on it. One solution is to remove the explicit dependency on log4j, or we could exclude the log4j dependency from slf4j. 

You can check this for yourself by running: 
```
mvn dependency:tree -Dverbose -Dincludes=log4j
```

We should go through each of these duplicate warnings, and resolve for each library.

**Comment by eshryane** [comment]
In this case (for log4j), the warning is harmless, as each included version is the same (1.2.17).

**Comment by eshryane** [comment]
Also - apache-log4j-extras itself includes duplicate classes from log4j

Ref. https://bz.apache.org/bugzilla/show_bug.cgi?id=55289

**Comment by eshryane** [comment]
Committed a fix on the branch log4j-duplicate-classes (200cdb27a1b8bc1f3ae266cc1043c7901c9edb74).

We need to test the impact of excluding the log4j dependency.

**Comment by eshryane** [comment]
- Merged log4j-duplicate-classes: added warning to root pom.xml (updated log4j-extras 1.2.17.1 not available in the mvn central repository).
- Committed fix for duplicated aspectj classes ddee736
- Committed fix for duplicated aopalliance classes c7abb4d
